### PR TITLE
subprocess mode error message

### PR DESCRIPTION
**Pull request by wrywerytwreywery**
This is a simple first go at fixing #54, prettifying error messages in subprocess mode.

### CODE REVIEW

**Comment by wrywerytwreywery** [comment]
The most recent commit implements suggestions for typos in commands.  Right now, defaults to making suggestions, and showing a max of 5, and it tries to sort suggestions in some kind of intelligent (but not computationally expensive) way.

Environment variables:
`SUGGEST_COMMANDS` is a boolean indicating whether suggestions should be made
`SUGGEST_ERROR_THRESHOLD` is an int indicating how "close" a command must be to consider it as a suggestion.
`SUGGEST_MAX_NUM` is an int indicating the maximum number of suggestions that should be shown.

Happy to make changes, or to roll back to the simpler error message in 4177c6bc068d5db4b8146b66a3c29df9a86aa717 if that is preferred.

File: xonsh/tools.py [hunk 1]
```diff
@@ -210,3 +210,38 @@ class redirect_stdout(_RedirectStream):
 class redirect_stderr(_RedirectStream):
     """Context manager for temporarily redirecting stderr to another file."""
     _stream = "stderr"
+
+
+# Modified from Public Domain code, by Magnus Lie Hetland
+# from http://hetland.org/coding/python/levenshtein.py 
+def levenshtein(a,b,max_dist=float('inf')):
```
Comment (scopatz) [inline xonsh/tools.py]: PEP8 spaces after commas

File: xonsh/built_ins.py [hunk 2]
```diff
@@ -351,8 +351,39 @@ def run_subproc(cmds, captured=True):
...
         else:
             stdin = prev_proc.stdout
-        proc = Popen(aliased_cmd, universal_newlines=uninew, env=ENV.detype(),
-                     stdin=stdin, stdout=stdout)
+        try:
+            proc = Popen(aliased_cmd, universal_newlines=uninew, env=ENV.detype(),
+                         stdin=stdin, stdout=stdout)
+        except FileNotFoundError:
+            cmd = aliased_cmd[0]
+            print('xonsh: subprocess mode: command not found: {0}'.format(cmd))
```
Comment (scopatz) [inline xonsh/built_ins.py]: Would you mind moving most of this block to its own function and the calling it from here? This function is already getting kinda long :)

**Comment by scopatz** [comment]
Frankly, this is incredible! I am so happy to see this.  Just a couple of minor comments to address.

**Comment by scopatz** [comment]
Thanks a ton @wrywerytwreywery

File: xonsh/built_ins.py (hunk 2 above)
Comment (wrywerytwreywery) [inline xonsh/built_ins.py]: No problem (I am assuming you mean mostly the suggestion stuff should move).  Am I correct in assuming that tools.py is a fine place to dump these kinds of things?  Or should I make a new suggester.py file or something like that?

Comment (scopatz) [inline xonsh/built_ins.py]: Ummhmm.  I think tools.py is fine.  If it gets much beyond a handful of functions, then a new module is worth it. For now, though, I don't think it is needed.

**Comment by scopatz** [comment]
Also @wrywerytwreywery - please ping me (by leaving a message on the issue) when you think this is ready for me to look at. I basically live off of github emailing me :)

**Comment by wrywerytwreywery** [comment]
Oh, sorry about that!  I think this is ready for a look now.

**Comment by scopatz** [comment]
This is *incredible*!  I can't thank you enough @wrywerytwreywery!

**Pull request merged**